tests/test_point_density.py

```diff
@@ -149,6 +160,52 @@ def test_point_density_ppf(scale: Scale):
     assert uniform_dist.ppf(1) == pytest.approx(scale.high, rel=0.1)
 
 
+@pytest.mark.parametrize(
+    "scale", [Scale(-3, 10), LogScale(0.01, 5, 500)],
+)
+def test_point_density_modes(scale: Scale):
+    dist = get_dist_from_scale(scale)
+
+    modes = dist.modes()
+
+    assert len(modes) > 0
+
+    mode_densities = np.array([dist.pdf(mode) for mode in modes])
+
+    assert np.all(mode_densities == mode_densities[0])
+
+    mode_density = mode_densities[0]
+
+    xs, _ = dist.to_arrays()
+
+    all_densities = np.array([dist.pdf(x) for x in xs])
+
+    assert mode_density == np.max(all_densities)
+
+
+@pytest.mark.parametrize(
+    "scale", [Scale(-3, 10), LogScale(0.01, 5, 500)],
+)
+def test_point_density_anti_modes(scale: Scale):
+    dist = get_dist_from_scale(scale)
+
+    anti_modes = dist.anti_modes()
+
+    assert len(anti_modes) > 0
+
+    anti_mode_densities = np.array([dist.pdf(anti_mode) for anti_mode in anti_modes])
+
+    assert np.all(anti_mode_densities == anti_mode_densities[0])
+
+    anti_mode_density = anti_mode_densities[0]
+
+    xs, _ = dist.to_arrays()
+
+    all_densities = np.array([dist.pdf(x) for x in xs])
+
+    assert anti_mode_density == np.min(all_densities)
```

Comment:
This code could be easier to read if it:

1. Used whitespace more intentionally
2. Had comments that explain what it is trying to do
3. Had less duplication

Sketch:

```py
dist = point_density_from_scale(scale)

extrema = (
    (dist.minima(), np.min),
    (dist.maxima(), np.max)
)

for extremal_xs, np_fn in extrema:

    # Check that the densities for all reported min/max values are the same
    extremal_densities = set(dist.pdf(x) for x in extremal_xs)
    assert len(extremal_densities) == 1
    extremal_density = extremal_densities.pop()

    # Check that they match the min/max we get from applying np min/max manually
    _, all_densities = dist.to_arrays()
    assert extremal_density == pytest.approx(np_fn(all_densities))
```